stonesoup/detector/beamformers.py

```diff
@@ -0,0 +1,505 @@
+# -*- coding: utf-8 -*-
+import numpy as np
+import math
+import random
+import copy
+from datetime import datetime, timedelta
+from stonesoup.base import Property, Base
+from stonesoup.buffered_generator import BufferedGenerator
+from stonesoup.models.measurement.linear import LinearGaussian
+from stonesoup.types.array import StateVector, CovarianceMatrix
+from stonesoup.types.detection import Detection
+
+
+def proposal(params, K, p_params):
```

Comment:
Are these standalone functions only used by `rjmcmc`? If so, I'd add them as class/static methods to the `rjmcmc` class.